src/lib.rs

```diff
@@ -53,6 +53,21 @@ use std::{
     time::{Duration, Instant},
 };
 
+/*
+I would highly suggest *not* using floats to represent time.  This is just a mess
+of issues due to variations on float accuracy and performance issues to be avoided.
+I would also suggest changing Nanoseconds to be:
+#[repr(transparent)]
+struct Nanoseconds {
+    duration: u64
+}
+This enforces type safety, allows impl From's for conversions and of course makes
+it compatible with C ABI's if you ever attempt to wrap this up and expose it to C.
+
+Also of note, I might suggest just using Microseconds as the exposed granularity
+since you are unlikely to ever get better than +-1 ms accuracy anyway.
+ */
```

Comment:
Floats are just an additional option, they are useful to have in some cases but you can always use `Duration`.

Reply:
First comment here, this was actually intended as a discussion between myself and someone else.  But, since I posted this to your repo (I believe the original author) on accident before it was intended, oops.  It would have gotten to you eventually but...

Anyway, I missed the variation with the Duration, that's all fine and better than a fixed type in general.  I have a knee jerk "don't use floats" reaction to timers as it has been nothing but a nightmare over the years of making games.  So, I avoid it at all times.